### DISCUSSION TITLE
Limiting User Creation and Device Addition per Domain in MeshCentral

### DISCUSSION THREAD

**Discussion by alanpereiracoelho** [Ylianst/MeshCentral#5649] [Q&A]
I am thrilled to have the opportunity to test this amazing tool, and I appreciate the work put into it. I have a question regarding user and device management: Is there a way to limit the number of users that can be created per domain and the number of devices that can be added? Thank you.

**Comment by si458** [comment]
yes `limits` under your `domains` section like below, give them a try and see if its what you need!
![image](https://github.com/Ylianst/MeshCentral/assets/765314/2f710b69-171f-4594-8189-b419562bcb4d)

**Reply by alanpereiracoelho** [reply]
would this setting be in the config.json?

**Reply by si458** [reply]
yes, you set it like so, a max of 1000 devices for the whole domain, and a max of 100 users can be created
![image](https://github.com/Ylianst/MeshCentral/assets/765314/b513fc8c-2294-48ed-8e89-32ac4c098ea9)

**Reply by alanpereiracoelho** [reply]
I am doing a test 'MaxDevices': 1,
It apparently worked, there is no message indicating that the limit has been exceeded... Is that correct?"

**Reply by si458** [reply]
erm yes weidly enough, you get no notification, if you wanted a notification of some sort, please open an enhancement request in issues
```
// Check if we already have too many devices for this domain
if (domain.limits && (typeof domain.limits.maxdevices == 'number')) {
  db.isMaxType(domain.limits.maxdevices, 'node', domain.id, function (ismax, count) {
    if (ismax == true) {
      // Too many devices in this domain.
      parent.agentStats.maxDomainDevicesReached++;
    } else {
      // We are under the limit, create the new device.
      completeAgentConnection2();
    }
  });
} else {
  completeAgentConnection2();
}
```

**Reply by alanpereiracoelho** [reply]
Where is this file so I can check the contents.

**Reply by si458** [reply]
im not sure i follow? the code i posted was inside `meshagent.js` line 753-765
but if you open up an enchancement request on github issues and people want this feature,
i can look at adding it in 👍

**Reply by alanpereiracoelho** [reply]
I wanted to let you know that I've followed your suggestion and opened a discussion on the MeshCentral GitHub. The discussion number is #5651. In this thread, I've addressed the issue of not receiving notifications when reaching the MaxDevices limit, along with a suggestion for a feature enhancement to improve user awareness and management of device limits.

I appreciate your guidance in this matter and would be delighted to hear your thoughts or any further suggestions on this topic.

Thanks again,
Alan
